src/kbc_modules/eaa_kbc/mod.rs

```diff
@@ -0,0 +1,223 @@
+        );
+
+        self.tcp_stream = Some(TcpStream::connect(&self.kbs_uri)?);
+
+        debug!("start negotiate (attestation) ...");
+        self.tls_handle
+            .as_ref()
+            .unwrap()
+            .negotiate(self.tcp_stream.as_ref().unwrap().as_raw_fd())
+            .map_err(|e| anyhow!("Negotiate Failed!:{:?}", e))?;
+
+        self.protocol_version = self.kbs_query_version()?;
+
+        Ok(())
+    }
+
+    fn kbs_query_version(&mut self) -> Result<String> {
+        let request = VersionRequest {
+            command: String::from("version"),
+        };
+        let trans_json = serde_json::to_string(&request)?;
+        let trans_data: &[u8] = trans_json.as_bytes();
+        let reciv_string: String = self.kbs_trans_and_reciv(trans_data, "Version")?;
+        let response: VersionResponse =
+            serde_json::from_str::<VersionResponse>(reciv_string.as_str())?;
+
+        match response.status.as_str() {
+            "OK" => return Ok(response.version),
+            "Fail" => return Err(anyhow!("The VersionResponse status is 'Fail'!")),
+            _ => return Err(anyhow!("Can't understand the VersionResponse status!")),
+        }
+    }
+
+    fn kbs_decrypt_payload(&mut self) -> Result<Vec<u8>> {
+        let request = DecryptionRequest {
+            command: String::from("Decrypt"),
+            blobs: vec![Blob {
+                kid: self.key_id.clone(),
+                encrypted_data: base64::encode(&self.encrypted_payload),
+                algorithm: "AES".to_string(),
+                key_length: 256,
+                iv: base64::encode(&self.iv),
+            }],
+        };
+        let trans_json = serde_json::to_string(&request)?;
+        let trans_data: &[u8] = trans_json.as_bytes();
+
+        let reciv_string: String = self.kbs_trans_and_reciv(trans_data, "Dcryption")?;
+        let response: DecryptionResponse =
+            serde_json::from_str::<DecryptionResponse>(reciv_string.as_str())?;
+
+        let payload_hashmap = match response.status.as_str() {
+            "OK" => response.data,
+            "Fail" => return Err(anyhow!(format!("Verdictd: {}", response.error.unwrap()))),
+            _ => return Err(anyhow!("Can't understand the DcryptionResponse status!")),
+        };
+
+        if let Some(hashmap_content) = payload_hashmap {
+            let encrypted_payload_string = base64::encode(&self.encrypted_payload);
+            let decrypted_payload_string = hashmap_content.get(&encrypted_payload_string).ok_or(anyhow!(
+                "There is no field matching the encrypted payload in the data field of DcryptionResponse"
+            ))?;
+            let decrypted_payload = base64::decode(decrypted_payload_string)?;
+            return Ok(decrypted_payload);
+        } else {
+            return Err(anyhow!(
+                "DecryptionResponse status is OK but the data is null!"
+            ));
+        }
+    }
+
+    fn kbs_get_kek(&mut self) -> Result<()> {
```

Comment:
It doesn't look like this function is called anywhere? I guess you are using the remote attestation approach in `kbs_decrypt_payload` where you send the wrapped key to the KBS. What is the point of this function then and what is the point of keeping a cache of the KEK if you never call this function?

Reply:
EAA KBS (Verdictd) can support both decrypting payload and getting KEK. We are not sure whether EAA KBC needs to handle decryption that can only use getting KEK in the future. Therefore, it is beneficial to retain this function. In order not to have an impact, it now exists as code comments.